### ISSUE TITLE
Issues with borders and focus indicators

### ISSUE THREAD

**Issue by this-username-system-sucks-ass** [guiodic/material-decoration#62]
After installing the version with the new focus indicators (#26), I noticed a couple issues. As I said in that thread I waited until I could confirm the issues on 6.5. I have it now and they are still here.

1. when using a transparent color scheme (i have a custom one but alpha 20 works well) and material decoration while there is a border, the opacity of the titlebar and border is 20% more than it should be. I confirmed this by changing the opacity I was using to be 20% lower, which made it look how it did before. 
2. while the window, or more specifically the titlebar and border, are transparent, the titlebar changes color with the border. based on guido's screenshots from #26, this is not intended.
3. The settings page doesn't very clearly show if the selected border colors will be transparent or what their respective opacities will be. this one is pretty minor but it would be nice to know for sure and be able to set it easier.

I did rebuild immediately after getting 6.5, so everything on my system should be up to date and representative of how the project is working. sorry to be kind of high maintenance with the focus indicator thing and thanks for the window decs.

**Comment by guiodic** [comment]
Can you take some screenshots?

**Comment by guiodic** [comment]
To be honest, I don't really understand the problem. I don't see the difference in opacity that you mention. You also write "which made it look how it did before." Before, but when exactly?
As for the colour, it's normal for the title bar to take on some colour from the border; that's how the kwin blur effect algorithm works.

**Comment by guiodic** [comment]
P.S. Just to be clear, the first screenshot is before the adjustment and the second is after you reduced the opacity by 20%?

**Comment by this-username-system-sucks-ass** [comment]
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/7658d4e2-64bc-4c52-93fe-f4e9940806fb" />

switched to a lighter background and bigger borders to show the difference more clearly. I also noticed that the border seems to be much lighter than the titlebar and actual window, which may be part of the issue.

**Comment by guiodic** [comment]
@this-username-system-sucks-ass 
As far as I understand, if you have a kwin border, it also uses the border for calculating the gradient.
What would be your desired result?

**Comment by guiodic** [comment]
<img width="815" height="424" alt="Image" src="https://github.com/user-attachments/assets/81d35b54-47e1-4a4c-ac9a-a90262622d8f" />

It seems to work now. But if you use Wayland, you need the [betterblur effect ](https://github.com/xarblu/kwin-effects-better-blur-dx) instead the standard one. The built-in blur effect on kwin_wayland does strange things, also without the modification I'm testing.

**Comment by guiodic** [comment]
I pushed in master an update, it makes an hole in the framebackground, this avoid the different color btwn border and the titlebar in itself

**Comment by this-username-system-sucks-ass** [comment]
It is currently failing to configure. after running the command [cmake .. -DQT_MAJOR_VERSION=6 -DQT_VERSION_MAJOR=6] it says it could not find epoxy, which I've verified is installed.

Here is the output:
`-- Found KF6: success (found version "6.23.0") found components: I18n
-- Found KF6: success (found version "6.23.0") found components: Config ConfigWidgets CoreAddons GuiAddons I18n IconThemes WindowSystem
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find epoxy (missing: epoxy_LIBRARIES epoxy_INCLUDE_DIRS)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/ECM/find-modules/Findepoxy.cmake:65 (find_package_handle_standard_args)
  /usr/share/cmake/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /usr/lib64/cmake/KWin/KWinConfig.cmake:50 (find_dependency)
  src/CMakeLists.txt:26 (find_package)


-- Configuring incomplete, errors occurred!
`

I'm not sure why it can't find epoxy or how to fix it. it may be an issue with fedora somehow since i've had build issues with this project on fedora before, but it's probably just me not knowing how to fix that. I can say at least that nothing should be broken since I reinstalled recently.
